pkg/cloudprovider/providers/azure/azure_instance_metadata.go

```diff
@@ -61,28 +54,43 @@ type Subnet struct {
 	Prefix  string `json:"prefix"`
 }
 
+type InstanceMetadata struct {
+	baseURL string
+}
+
+func NewInstanceMetadata() *InstanceMetadata {
+	return &InstanceMetadata{
+		baseURL: metadataURL,
+	}
+}
+
+// makeMetadataURL makes a complete metadata URL from the given path.
+func (i *InstanceMetadata) makeMetadataURL(path string) string {
+	return i.baseURL + path
+}
+
 // QueryMetadataJSON queries the metadata server and populates the passed in object
-func QueryMetadataJSON(path string, obj interface{}) error {
-	data, err := queryMetadataBytes(path, "json")
+func (i *InstanceMetadata) QueryMetadataJSON(path string, obj interface{}) error {
```

Comment:
It'd be nice to incorporate into the name of this method the idea that it mutates the passed in object interface. `QueryMetadataJSONUnmarshall`? (ugh)